File: src/config/nim.rs
use serde::{Deserialize, Serialize};

use crate::config::constants::ANTHROPIC_DEFAULT_MAX_OUTPUT_TOKENS;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NimSettings {
    pub temperature: f64,
    pub top_p: f64,
    pub top_k: i32,
    pub max_tokens: i32,
    pub presence_penalty: f64,
    pub frequency_penalty: f64,
    pub min_p: f64,
    pub repetition_penalty: f64,
    pub seed: Option<i64>,
    pub stop: Option<String>,
    pub parallel_tool_calls: bool,
    pub ignore_eos: bool,
    pub min_tokens: i32,
    pub chat_template: Option<String>,
    pub request_id: Option<String>,
}

impl Default for NimSettings {
    fn default() -> Self {
        Self {
            temperature: 1.0,
            top_p: 1.0,
            top_k: -1,
            max_tokens: ANTHROPIC_DEFAULT_MAX_OUTPUT_TOKENS,
            presence_penalty: 0.0,
            frequency_penalty: 0.0,
            min_p: 0.0,
            repetition_penalty: 1.0,
            seed: None,
            stop: None,
            parallel_tool_calls: true,
            ignore_eos: false,
            min_tokens: 0,
            chat_template: None,
            request_id: None,
        }
    }
}

File: config/nim.py
"""NVIDIA NIM specific configuration."""

from pydantic import BaseModel


class NimSettings(BaseModel):
    """Settings specific to NVIDIA NIM provider."""

    temperature: float | None = 0.7
    top_p: float | None = 0.9
    top_k: int | None = 50
    max_tokens: int | None = 4096
    stop: list[str] | None = None
    presence_penalty: float = 0.0
    frequency_penalty: float = 0.0
    seed: int | None = None
    parallel_tool_calls: bool = True
    min_p: float = 0.0
    repetition_penalty: float = 1.0
    min_tokens: int = 0
    chat_template: str | None = None
    request_id: str | None = None
    return_tokens_as_token_ids: bool = False
    include_stop_str_in_output: bool = False
    ignore_eos: bool = False
    reasoning_effort: str | None = None
    include_reasoning: bool = True
